joy_teleop/joy_teleop/joy_teleop.py

```diff
@@ -317,21 +317,20 @@ def update_actions(self, evt=None):
                 self.register_action(name, cmd)
 
 
-def main():
-    try:
-        rclpy.init()
-
-        node = JoyTeleop()
+def main(args=None):
+    rclpy.init(args=args)
+    node = JoyTeleop()
 
+    try:
         rclpy.spin(node)
-
-        node.destroy_node()
-        rclpy.shutdown()
-    except JoyTeleopException:
-        pass
+    except JoyTeleopException as e:
+        node.get_logger().error(e.message)
     except KeyboardInterrupt:
         pass
 
+    node.destroy_node()
+    rclpy.shutdown()
```

Comment:
Genuine question, aren't those two implicitly called? 
```python
node.destroy_node()
rclpy.shutdown()
```

Reply:
I'm not sure this is relevant enough to hold up this PR. Those lines were they part of the original code as well.

Reply:
It is not, that's why I started the sentence with 'Genuine question' ;)